src/boot/efi/xbootldr.c

```diff
@@ -149,11 +133,18 @@ static EFI_STATUS try_gpt(
                 if (end < start) /* Bogus? */
                         continue;
 
-                ret_hd->PartitionNumber = i + 1;
-                ret_hd->PartitionStart = start;
-                ret_hd->PartitionSize = end - start + 1;
-                ret_hd->MBRType = MBR_TYPE_EFI_PARTITION_TABLE_HEADER;
-                ret_hd->SignatureType = SIGNATURE_TYPE_GUID;
+                *ret_hd = (HARDDRIVE_DEVICE_PATH) {
+                        .Header = (EFI_DEVICE_PATH) {
```

Comment:
you can drop the `(EFI_DEVICE_PATH)` here. The type is clear. In fact, the fact you can specify it here at all is a GNU exension...